sdk/core/core/inc/az_platform.h

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef _az_PLATFORM_H
+#define _az_PLATFORM_H
+
+#include <az_result.h>
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <_az_cfg_prefix.h>
+
+/**
+ * @brief az_platform_clock_msec gets platform clock in milliseconds.
+ * The moment of time where clock starts is undefined, but if this function is getting called twice
+ * with one second interval, the difference between the values returned should be equal to 1000.
+ * @return Platform clock in milliseconds.
+ */
+AZ_NODISCARD int64_t az_platform_clock_msec();
+
+/**
+ * @brief az_platform_sleep_msec tells platform to sleep for a given number of milliseconds.
+ * @param milliseconds number of milliseconds to sleep.
+ * @remarks For passing \p milliseconds with negative values and 0, behavior is undefined.
+ */
+void az_platform_sleep_msec(int32_t milliseconds);
+
+/**
+ * @brief az_platform_atomic_compare_exchange Conditionally exchanges values of two pointers in a
+ * thread-safe manner.
+ *
+ * @param obj a pointer to a pointer that needs to be changed.
+ * @param expected an expected value of a value that \p obj is pointing to, prior to exchange.
+ * @param desired a value to assign to the value that is pointed by \p obj, if its value equals to
```

Comment:
Capitalize. 

nit: Also, I am having difficulty understanding this a bit due to the repetition. Maybe there is a clearer way to explain the parameters.

Reply:
I'll try to see if I can rephrase this, maybe I'll borrow something from https://docs.microsoft.com/en-us/windows/win32/api/winnt/nf-winnt-interlockedcompareexchangepointer. Ah, documenting the code you just written feels a bit like playing chess against yourself :)

Reply:
I capitalized and tried to rephrase, resolving the conversation thread.